api/nivs.js

```diff
@@ -0,0 +1,126 @@
+import { backendsPromise, visualisationApiUrl } from '~/api/backends/'
+import axios from 'axios'
+import { instanceId } from '~/constants/localUUIDs'
+
+const stateFileName = 'state.json'
+const builderId = 'a734e3e7-ca10-41f2-9638-a19710d6430d'
+
+export async function setupSyncStore() {
+  return await Promise.all([getPresignedURLforGET(), getPresignedURLforPUT()])
+}
+
+export async function getInstance() {
+  await backendsPromise
+  return axios.get(`${visualisationApiUrl}/instances/${instanceId}`)
+}
+
+function getStateFileUrl(listOfFiles) {
+  const correctFile = listOfFiles.find(e => e.file_name === stateFileName)
+  if (correctFile) {
+    return correctFile.presigned_url
+  }
+  return null
+}
+
+export async function getPresignedURLforGET() {
+  await backendsPromise
+  const response = await axios.get(
+    `${visualisationApiUrl}/instances/${instanceId}/state-sync`
+  )
+  return getStateFileUrl(response.data)
+}
+
+export async function getPresignedURLforPUT() {
+  await backendsPromise
+  const response = await axios.post(
+    `${visualisationApiUrl}/instances/${instanceId}/state-sync`,
+    {
+      files: [stateFileName],
+    }
+  )
+  return getStateFileUrl(response.data)
+}
+
+export async function downloadPlot(plotId) {
+  await backendsPromise
+  const response = await axios.get(`${visualisationApiUrl}/plots/${plotId}`)
+  const presignedUrl = response.data.presigned_urls[0].presigned_url
+  const getResponse = await axios.get(presignedUrl, {
+    headers: {
+      'Content-Type': 'image/png',
+    },
+  })
+  return getResponse.data
```

Comment:
Think you've mentioned it but I guess there's stuff in here that might be useful in the webapp itself?

Reply:
Yes lots of code that could be common